src/main/scala/is/hail/expr/ir/ArraySorter.scala

```diff
@@ -80,7 +92,9 @@ class ArraySorter(mb: EmitMethodBuilder, array: StagedArrayBuilder, keyOnly: Boo
     }
 
     def lt(i: Code[Int], j: Code[Int]): Code[Boolean] =
-      compare(region, (array.isMissing(i), array(i)), region, (array.isMissing(j), array(j))) < 0
+      asc.mux(
+        less(region, (array.isMissing(i), array(i)), region, (array.isMissing(j), array(j))),
+        greater(region, (array.isMissing(i), array(i)), region, (array.isMissing(j), array(j))))
```

Comment:
Most of the changes in ArraySorter are just to add the ascending flag. This is the only place that looks at the flag.